lightning/src/ln/channelmanager.rs

```diff
@@ -3557,8 +3468,8 @@ where
 
 		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
-		let (chan, msg) = match peer_state.outbound_v1_channel_by_id.remove(temporary_channel_id) {
-			Some(chan) => {
+		let (chan, msg) = match peer_state.channel_by_id.remove(temporary_channel_id) {
```

Comment:
Unfortunately we must take ownership so that `chan.get_funding_created` can receive `self`. Otherwise we can't get to the context to move it. There's also no Default implementation of `ChannelContext` to do a `mem::take` to take it through a `&mut self` when we construct the `Channel`. Creating such a default implementation doesn't seem practical. Also, creating another layer of Option for `context` in a channel would make things messy.

Not sure what's best here. The part I probably like least is that we need to re-add channels that are not `OutboundV1Channel`s.

Reply:
Larger is ok and it's not that much larger. I was being really silly. `entry` works :100: